core/domain/exp_domain.py

```diff
@@ -1618,12 +2141,22 @@ def from_dict(
 
     @classmethod
     def _require_valid_state_name(cls, name):
+        """Validates name string.
+
+        Args:
+            name: str. The name to validate.
+        """
         utils.require_valid_name(name, 'a state name')
 
     def validate(self, strict=False):
-        """Validates the exploration before it is committed to storage.
+        """Validates various properties of the Exploration.
+
+        Args:
+            strict: bool. If True more properties are validated.
```

Comment:
If True, the exploration is assumed to be published, and the validation checks are stricter.